test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/RequestTests.cs

```diff
@@ -520,6 +521,53 @@ public void CanUpgradeRequestWithConnectionKeepAliveUpgradeHeader()
             }
         }
 
+        [Theory]
```

Comment:
It already returns `string.Empty` if not set. We can't throw on set because it's set by middleware (`UsePathBaseMiddleware` in HttpAbstractions).